src/redux.js

```diff
@@ -739,65 +694,38 @@ export function getSelectedCategoricalFeatures(state) {
   return intersection;
 }
 
-export function getSelectedNumericalColumns(state) {
-  let intersection = getNumericalColumns(state).filter(
-    x => state.selectedFeatures.includes(x) || x === state.labelColumn
-  );
-  return intersection;
-}
-
 export function getSelectedNumericalFeatures(state) {
   let intersection = getNumericalColumns(state).filter(x =>
     state.selectedFeatures.includes(x)
   );
   return intersection;
 }
 
-export function getNumericalColumns(state) {
-  return filterColumnsByType(state, ColumnTypes.NUMERICAL);
-}
-
-/*
-  Categorical columns with too many unique values are unlikley to make
-  accurate models, and we don't want to overflow the metadata column for saved
-  models.
-*/
-function hasTooManyUniqueOptions(state, column) {
-  if (isColumnCategorical(state, column)) {
-    const uniqueOptionsCount =
-      getUniqueOptions(state, state.currentColumn).length;
-    return uniqueOptionsCount > UNIQUE_OPTIONS_MAX;
-  }
-  return false;
-}
-
-export function getSelectableFeatures(state) {
-  return getFeatures(state).filter(
-    column =>
-      column !== state.labelColumn && !state.selectedFeatures.includes(column)
+export function getSelectedNumericalColumns(state) {
+  let intersection = getNumericalColumns(state).filter(
+    x => state.selectedFeatures.includes(x) || x === state.labelColumn
   );
+  return intersection;
 }
 
-export function getSelectableLabels(state) {
-  return getFeatures(state).filter(x => !state.selectedFeatures.includes(x));
+function getSelectedColumns(state) {
+  return state.selectedFeatures
+    .concat(state.labelColumn)
+    .filter(column => column !== undefined && column !== "")
+    .map(columnId => {
+      return { id: columnId, readOnly: isColumnReadOnly(state, columnId) };
+    });
 }
 
-export function getUniqueOptions(state, column) {
-  return Array.from(new Set(state.data.map(row => row[column]))).filter(
-    option => option !== undefined && option !== ""
-  );
-}
+/* Functions for getting specific details about a batch of columns.  */
```

Comment:
Could the functions below this comment go into the `columnDetails.js` file?

Reply:
The distinction I made between the functions in `columnDetails` and the functions in this section are that `columnDetails` functions return information about a single specified column, whereas those in this section of redux.js are about multiple columns as a collection. In general, I kept anything that is being called directly from a UI component in the redux.js file so that it was accessible via connect.

Reply:
Makes sense, thank you!